tests/PolyType.Tests/Utilities/AggregatingTypeShapeProviderTests.cs

```diff
@@ -0,0 +1,185 @@
+        providers[0] = new MockTypeShapeProvider
+        {
+            Shapes =
+            {
+                [typeof(int)] = new MockTypeShape<int>(),
+            },
+        };
+
+        // Verify that the new provider is not used.
+        Assert.Null(aggregate.GetShape(typeof(int)));
+    }
+
+    [Fact]
+    public void IterateOverProviders()
+    {
+        AggregatingTypeShapeProvider provider = new(new MockTypeShapeProvider());
+        Assert.Single(provider.Providers);
+    }
+
+    /// <summary>
+    /// Validates that the aggregating provider does not cache results.
+    /// </summary>
+    /// <remarks>
+    /// Whether it caches or not is something of an arbitrary design decision.
+    /// This test is to ensure that the behavior is documented and intentional.
+    /// </remarks>
+    [Fact]
+    public void CachePolicy_ShapeNotCached()
+    {
+        MockTypeShapeProvider first = new()
+        {
+            Shapes =
+            {
+                [typeof(int)] = new MockTypeShape<int>(),
+            },
+        };
+        AggregatingTypeShapeProvider aggregate = new(first);
+        Assert.Same(first.Shapes[typeof(int)], aggregate.GetShape(typeof(int)));
+
+        // Remove the shape, and verify that the aggregator no longer returns it.
+        first.Shapes.Remove(typeof(int));
+        Assert.Null(aggregate.GetShape(typeof(int)));
+    }
+
+    /// <summary>
+    /// Validates that the aggregating provider does not cache the provider that was used previously.
+    /// </summary>
+    /// <remarks>
+    /// Whether it caches or not is something of an arbitrary design decision.
+    /// This test is to ensure that the behavior is documented and intentional.
+    /// </remarks>
+    [Fact]
+    public void CachePolicy_ShapeProviderNotCached()
+    {
+        MockTypeShapeProvider first = new()
+        {
+            Shapes =
+            {
+                [typeof(int)] = new MockTypeShape<int>(),
+            },
+        };
+        MockTypeShapeProvider second = new()
+        {
+            Shapes =
+            {
+                [typeof(string)] = new MockTypeShape<string>(),
+            },
+        };
+        AggregatingTypeShapeProvider aggregate = new(first, second);
+        Assert.NotNull(aggregate.GetShape(typeof(string)));
+
+        // Add a new string shape to the first provider.
+        first.Shapes[typeof(string)] = new MockTypeShape<string>();
+
+        // The aggregator should return the new shape.
+        Assert.Same(first.Shapes[typeof(string)], aggregate.GetShape(typeof(string)));
+    }
+
+    private class MockTypeShapeProvider : ITypeShapeProvider
+    {
+        internal Dictionary<Type, ITypeShape> Shapes { get; } = new();
+
+        public ITypeShape? GetShape(Type type) => Shapes.TryGetValue(type, out ITypeShape? shape) ? shape : null;
+    }
+
+    private class MockTypeShape<T> : ITypeShape<T>
+    {
+        public Type Type => throw new NotImplementedException();
```

Comment:
Not sure why Codecov is giving out coverage reports for test projects -- I thought I'd disabled that.

Reply:
I added an attribute to exclude it.